Test fails on dev machine.

master, OSX, gnovm test fails.

```
gnovm % go test ./cmd/... -v -run TestBuild/invalid_go_files
=== RUN   TestBuild
=== RUN   TestBuild/invalid_go_files
=== PAUSE TestBuild/invalid_go_files
=== CONT  TestBuild/invalid_go_files
    testscript.go:534: WORK=$WORK
        PATH=/usr/local/go/bin:/Library/Frameworks/Python.framework/Versions/3.11/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/Users/jk/gopath/bin:/usr/local/go/bin
        GOTRACEBACK=system
        HOME=/var/folders/19/85srrz214qlbd122_0jzskvw0000gn/T/TestBuild3366989911/002
        TMPDIR=$WORK/.tmp
        devnull=/dev/null
        /=/
        :=:
        $=$
        GOCOVERDIR=
        exe=
        GNOROOT=/Users/jk/gopath/src/github.com/gnolang/gno
        
        # Run gno build with invalid go files (0.727s)
        > ! gno build .
        [stderr]
        command-line-arguments
        # command-line-arguments
        ./file1.go:3:1: syntax error: non-declaration statement outside function body
        
        ./.: build pkg: std go compiler: exit status 2
        
        command-line-arguments
        # command-line-arguments
        sub/file2.go:3:1: syntax error: non-declaration statement outside function body
        
        ./sub: build pkg: std go compiler: exit status 2
        
        2 go build errors
        
        [exit status 1]
        > ! stdout .+
        > stderr '\./file1\.go:3:1: syntax error: non-declaration statement outside function body'
        > stderr '\./\.: build pkg: std go compiler: exit status 1'
        FAIL: testdata/gno_build/invalid_go_files.txtar:7: no match for `\./\.: build pkg: std go compiler: exit status 1` found in stderr
        
--- FAIL: TestBuild (0.90s)
    --- FAIL: TestBuild/invalid_go_files (0.73s)
```

Command to test this behavior using directly the go command:

```
 $ rm -rf /tmp/xxx && go run github.com/rogpeppe/go-internal/cmd/txtar-x@latest -C /tmp/xxx ./gnovm/cmd/gno/testdata/gno_build/invalid_go_files.txtar && go build -C /tmp/xxx .; echo $?
```

Which returns :
```
# gnobuild
./file1.go:3:1: syntax error: non-declaration statement outside function body
1
```

We can see the exit code is 1 here.

--- a/gnovm/cmd/gno/testdata/gno_build/invalid_go_files.txtar
+++ b/gnovm/cmd/gno/testdata/gno_build/invalid_go_files.txtar
@@ -4,9 +4,9 @@
 
 ! stdout .+
 stderr '\./file1\.go:3:1: syntax error: non-declaration statement outside function body'
-stderr '\./\.: build pkg: std go compiler: exit status 1'
+stderr '\./\.: build pkg: std go compiler'
 stderr 'sub/file2\.go:3:1: syntax error: non-declaration statement outside function body'
-stderr '\./sub: build pkg: std go compiler: exit status 1'
+stderr '\./sub: build pkg: std go compiler'
 
 -- go.mod --
 module gnobuild

--- a/gnovm/cmd/gno/testdata/gno_build/no_go_files.txtar
+++ b/gnovm/cmd/gno/testdata/gno_build/no_go_files.txtar
@@ -4,8 +4,8 @@
 
 ! stdout .+
 stderr -count=2 'no Go files in '$WORK
-stderr '\./\.: build pkg: std go compiler: exit status 1'
-stderr '\./sub: build pkg: std go compiler: exit status 1'
+stderr '\./\.: build pkg: std go compiler'
+stderr '\./sub: build pkg: std go compiler'
 
 -- go.mod --
 module gnobuild

--- a/gnovm/cmd/gno/testdata/gno_build/no_gomod.txtar
+++ b/gnovm/cmd/gno/testdata/gno_build/no_gomod.txtar
@@ -4,8 +4,8 @@
 
 ! stdout .+
 stderr -count=2 'go: go.mod file not found in current directory or any parent directory'
-stderr './.: build pkg: std go compiler: exit status 1'
-stderr './sub: build pkg: std go compiler: exit status 1'
+stderr './.: build pkg: std go compiler'
+stderr './sub: build pkg: std go compiler'
 
 -- main.gno --
 package main

